test/blackbox-tests/test-cases/no_dynlink.t

```diff
@@ -0,0 +1,32 @@
+This test checks that if a library is declared with `(no_dynlink)`, then the
+corresponding `.cmxs` file is *not* built.
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.17)
+  > EOF
+
+First we check the behaviour when `(no_dynlink)` is not present.
+
+  $ cat >dune <<EOF
+  > (library
+  >  (name mylib))
+  > EOF
+
+  $ touch a.ml
+
+  $ dune build _build/default/mylib.cmxs
+
+Now with `(no_dynlink)`.
+
+  $ cat >dune <<EOF
+  > (library
+  >  (name mylib)
+  >  (no_dynlink))
+  > EOF
+
+  $ dune clean
+
+  $ dune build _build/default/mylib.cmxs
```

Comment:
Could you add some tests for the .install file as well?

Reply:
Good idea. Fixed, thanks.